pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -804,6 +804,16 @@ protected void handleProducerRemoved(Producer producer) {
                 if (ex.getCause() instanceof ConsumerBusyException) {
                     log.warn("[{}][{}] Consumer {} {} already connected", topic, subscriptionName, consumerId,
                             consumerName);
+                    Consumer consumer = null;
+                    try {
+                        consumer = subscriptionFuture.isDone() ? getActiveConsumer(subscriptionFuture.get()) : null;
+                        // cleanup consumer if connection is already closed
+                        if (consumer != null && !consumer.cnx().isActive()) {
+                            consumer.close();
+                        }
+                    } catch (Exception be) {
+                        log.info("Failed to clean up consumer on closed connection {}, {}", consumer, be.getMessage());
```

Comment:
Any reason this log is info level?

Reply:
it's just info and doesn't have to be critical. therefore, log level is info.

Reply:
I just want to make sure that if it's an exception goes here, the problem you mentioned won't occur,
as "subsequent consumer creation requests are keep failing" is totally worth to be an error.

Reply:
To me. It seems like a bandaid. I think we should see how we landed in this situation. Perhaps we should have clean the consumer at the connection close itself. If there is a race condition we should the race condition itself.

Reply:
@Jason918 changed to error log level.

@pkumar-singh this is part handles resiliency to avoid consumer unavailability due to any kind of race condition.